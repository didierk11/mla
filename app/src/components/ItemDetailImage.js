import React from "react";

const ItemDetailImage = (props) => {
  const src = props.src;
  const alt = props.alt;

  return (
    <div>
      <div className="item-detail-img">
        <img src={src} alt={alt}></img>
      </div>
    </div>
  );
};

export default ItemDetailImage;
