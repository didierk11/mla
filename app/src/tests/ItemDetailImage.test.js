import React from "react";
import { render } from "react-dom";
import ItemDetailImage from "../components/ItemDetailImage";
import renderer from "react-test-renderer";
import data from "./dataSets/items.json";

//testing data
const src = data.src;
const alt = data.title;

describe("Snapshot test:", () => {
  it("ItemDetailImage render correctly", () => {
    const div = renderer
      .create(<ItemDetailImage src={src} alt={alt} />)
      .toJSON();
    expect(div).toMatchSnapshot();
  });
});

describe("Smoke test:", () => {
  it("ItemDetailImage render without crashing", () => {
    const div = document.createElement("div");
    render(<ItemDetailImage src={src} alt={alt} />, div);
  });
});
