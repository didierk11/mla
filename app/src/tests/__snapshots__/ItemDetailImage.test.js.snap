// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test: ItemDetailImage render correctly 1`] = `
<div>
  <div
    className="item-detail-img"
  >
    <img
      alt="Pelota De Futbol Nº 5 + Calidad 350 Gramos Cámara De Látex"
      src="https://http2.mlstatic.com/D_647277-MLA42892938048_072020-O.jpg"
    />
  </div>
</div>
`;
